src/flashFormatter/C33FlashFormatter.cpp

```diff
@@ -71,4 +68,60 @@ bool C33FlashFormatter::formatPartition() {
   return true;
 }
 
+bool C33FlashFormatter::checkCACertificatesPartition()
```

Comment:
I feel it would be better to add a slightly modified version of `checkCACertificatesPartition` and `flashCACertificates` to the base class and reuse the same code for the H7 and C33. Do you agree?

Reply:
Considering adding new interfaces to the base class like `checkCACertificatesPartition(MBRBlockDevice bd, String base_path)` and `flashCACertificates(MBRBlockDevice bd, String base_path)`, I don't think it's possible to move the `checkCACertificatesPartition` and `flashCACertificates` to the base class for two reasons:

- the base class is used for all other architectures, so they could not have the  `FATFileSystem`  and `MBRBlockDevice` lib.
- moving the `flashCACertificates` to the base class it requires to include the *certificates.h* file for every architectures and this could cause an increase of storage usage on every targets